Support for homebrew by creating published releases. (issue #7)

eddiewebb opened the pull request:
This PR  addressed issue #7 and only modifies the config.yml to add a GitHub tarball that is versioned using version from script (matches SNAP versioning).

To be included back into CircleCI repo, a few changes are needed:

1. Uncomment the SNAP build and deploy steps which will not pass on my local repo (i can do this before merging, but will break build on my side)
2. Add `GITHUB_TOKEN` as circleci environment variable with a `public_repos` access token for circleCI repos to be able to publish release.



```
$ brew install eddiewebb/circleci/circleci
==> Tapping eddiewebb/circleci
Cloning into '/Users/eddie/homebrew/Library/Taps/eddiewebb/homebrew-circleci'...
remote: Counting objects: 7, done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 7 (delta 0), reused 3 (delta 0), pack-reused 0
Unpacking objects: 100% (7/7), done.
Tapped 1 formula (33 files, 23.9KB)
==> Installing circleci from eddiewebb/circleci
==> Downloading https://github.com/eddiewebb/local-cli/releases/download/v0.0.4705-deba4df/circleci-0.0.4705-deba4df.tar.gz
Already downloaded: /Users/n0158588/Library/Caches/Homebrew/circleci-0.0.4705.tar.gz
🍺  /usr/local/Cellar/circleci/0.0.4705: 3 files, 5.0KB, built in 1 second
$ circleci --version
0.0.4705-deba4df

```

felicianotech commented:
I don't have time to review this today but confirming that I've seen this. This is definitely a need so will be looking forward to getting this PR merged.

felicianotech commented on .circleci/config.yml:
```diff
@@ -21,12 +27,29 @@ jobs:
+      - run:
+          name: Generate tarball
+          command: |
+            VERSION=$(bash circleci.sh --version)
```
As with the Git tag created later on, can we prefix all versions with the letter 'v'? The character doesn't have to been in the envar itself but in strings when it's concatenated in them.

felicianotech commented on .circleci/config.yml:
```diff
@@ -38,3 +61,19 @@ jobs:
+            go get github.com/tcnksm/ghr
+            VERSION=$(ls *.tar.gz | sed -e 's/circleci-\([a-z0-9.-]*\)\.tar\.gz/\1/')
+            echo "Pushing version ${VERSION} to github releases"
+            ghr -t ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} -delete v${VERSION} circleci-${VERSION}.tar.gz
```
We might as well publish the `.snap` file to GitHub here as well.

eddiewebb commented:
Latest commit addresses feedback except filename of snap which is controlled by snapcraft version. https://github.com/eddiewebb/local-cli/releases.

publish-snap fails since my CCI does nt have store credentials.

felicianotech commented:
Cool. LGTM except for the failed job that we expect to fail.
